drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/test/java/org/drools/workbench/screens/guided/rule/client/widget/FromAccumulateCompositeFactPatternWidgetTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.rule.client.widget;
+
+import com.google.gwt.dom.client.AnchorElement;
+import com.google.gwt.event.shared.EventBus;
+import com.google.gwt.user.client.ui.RootPanel;
+import com.google.gwtmockito.GwtMockito;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import com.google.gwtmockito.WithClassesToStub;
+import org.drools.workbench.models.datamodel.rule.FromAccumulateCompositeFactPattern;
+import org.drools.workbench.models.datamodel.rule.RuleModel;
+import org.drools.workbench.screens.guided.rule.client.editor.RuleModeller;
+import org.drools.workbench.screens.guided.rule.client.resources.GuidedRuleEditorResources;
+import org.gwtbootstrap3.client.ui.ListBox;
+import org.gwtbootstrap3.client.ui.html.Text;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.soup.project.datamodel.imports.Import;
+import org.kie.soup.project.datamodel.imports.Imports;
+import org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracle;
+import org.mockito.Mock;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@WithClassesToStub({RootPanel.class, Text.class, AnchorElement.class})
+@RunWith(GwtMockitoTestRunner.class)
+public class FromAccumulateCompositeFactPatternWidgetTest {
+
+    private String[] factTypes;
+
+    private RuleModel ruleModel;
+
+    @Mock
+    private AsyncPackageDataModelOracle oracle;
+
+    @Mock
+    private RuleModeller ruleModeller;
+
+    @Mock
+    private EventBus eventBus;
+
+    @Mock
+    private FromAccumulateCompositeFactPattern pattern;
+
+    @Mock
+    private ListBox listBox;
+
+    private FromAccumulateCompositeFactPatternWidget fromAccumulateWidget;
+
+    @Before
+    public void setUp() throws Exception {
+        GwtMockito.useProviderForType(ListBox.class, aClass -> listBox);
+
+        ruleModel = new RuleModel();
+        factTypes = new String[2];
+        factTypes[0] = "Person";
```

Comment:
If this is supposed to be the same `Person` as on line 93, please make it a constant and share it. Don't expect people to understand what you mean, make the code say what you mean.